src/main/java/com/conveyal/gtfs/validator/FlexValidator.java

```diff
@@ -412,18 +532,32 @@ public static List<NewGTFSError> validateBookingRule(BookingRule bookingRule) {
     }
 
     /**
-     * Check if a stop area or location matches any stop ids.
+     * Check if a location group id or location id matches any stop ids.
      */
-    private static boolean stopAreaOrLocationIsStop(List<Stop> stops, String id) {
-        return !stops.isEmpty() && stops.stream().anyMatch(stop -> stop.stop_id.equals(id));
+    private static boolean hasMatchOnStopId(List<Stop> stops, String id) {
+        return
+            stops != null &&
+            !stops.isEmpty() &&
+            stops.stream().anyMatch(stop -> stop.stop_id != null && stop.stop_id.equals(id));
     }
 
     /**
-     * Check if a stop area (area id) matches any locations.
+     * Check if a location group id matches any location ids.
      */
-    private static boolean stopAreaIsLocation(List<Location> locations, String areaId) {
-        return !locations.isEmpty() &&
-            locations.stream().anyMatch(location -> location.location_id.equals(areaId));
+    private static boolean hasMatchOnLocationId(List<Location> locations, String id) {
+        return
+            !locations.isEmpty() &&
+            locations.stream().anyMatch(location -> location.location_id != null && location.location_id.equals(id));
```

Comment:
You might be able to reverse the equality check `id.equals(...)` and remove the extra null check.

Reply:
Null check is still required.